src/graphics/graphics-device.js

```diff
@@ -3391,11 +3409,25 @@ class GraphicsDevice extends EventHandler {
 
         // Check for errors
         if (!gl.getShaderParameter(glVertexShader, gl.COMPILE_STATUS)) {
-            console.error("Failed to compile vertex shader:\n\n" + this._addLineNumbers(definition.vshader) + "\n\n" + gl.getShaderInfoLog(glVertexShader));
+            const infoLog = gl.getShaderInfoLog(glVertexShader);
+            const [code, error] = this._processError(definition.vshader, infoLog);
+            // #if _DEBUG
+            error.shader = shader;
+            console.error(`Failed to compile vertex shader:\n\n${infoLog}\n${code}`, error);
+            // #else
+            console.error(`Failed to compile vertex shader:\n\n${infoLog}\n${code}`);
+            // #endif
```

Comment:
Could you also wrap this block in a function so that we don't have this pretty much identical code twice please.

Reply:
perhaps even 'getShaderParameter' line